Demos/Raven.NET.Demo.WebApi/Controllers/OrderController.cs

```diff
@@ -0,0 +1,87 @@
+using Microsoft.AspNetCore.Mvc;
+using Raven.NET.Demo.WebApi.Model;
+using Raven.NET.Demo.WebApi.Repositories.Interfaces;
+using Raven.NET.Demo.WebApi.RequestModel;
+using Raven.NET.Demo.WebApi.Services.Interfaces;
+
+namespace Raven.NET.Demo.WebApi.Controllers
+{
+//PS: This api is not an example of properly done SOLID Web Api, its simple showcase to see how Raven.NET components can be implemented
+    [ApiController]
+    [Route("api/[controller]")]
+    public class OrderController : ControllerBase
+    {
+        private readonly ILogger<OrderController> _logger;
+        private readonly IOrderRepository _orderRepository;
+        private readonly ICustomerRepository _customerRepository;
+        private readonly IOrderUpdateService _orderUpdateService;
+
+        public OrderController(ILogger<OrderController> logger, IOrderRepository orderRepository,
+            ICustomerRepository customerRepository, IOrderUpdateService orderUpdateService)
+        {
+            _logger = logger;
+            _orderRepository = orderRepository;
+            _customerRepository = customerRepository;
+            _orderUpdateService = orderUpdateService;
+        }
+
+        // GET: api/Order
+        [HttpGet]
+        public IEnumerable<Order> Get()
+        {
+            _logger.LogInformation("Get all orders invoked");
+            var result = _orderRepository.GetAll();
+            return result;
+        }
+
+        // GET: api/Order/8d35a21b-2b53-4407-8959-75910582af36
+        [HttpGet("{id}")]
+        public Order Get(Guid id)
+        {
+            _logger.LogInformation($"Get order {id} invoked");
+            var result = _orderRepository.Get(id);
+            return result;
+        }
+
+        // POST: api/Order
+        [HttpPost]
+        public IActionResult Post([FromBody] OrderCreateRequest value)
+        {
+            _logger.LogInformation($"Create order {value.Number} invoked");
+            var customer = _customerRepository.Get(value.CustomerId);
+            Order order = new Order(value.Number, customer, value.Product, value.Price);
+            customer.OrderCount++;
+            
+            //We can add additional logic there for example to keep track only on specified orders.
+            _orderUpdateService.Track(order);
+            _orderUpdateService.ProcessOrder(order);
```

Comment:
Move Track/ProcessOrder after saving - on first run it tries to check for non existing order, causes exception. Not critical but worth looking into it. And double check it if it's actually relevant

Reply:
Done